src/library/trackset/baseplaylistfeature.cpp

```diff
@@ -536,6 +536,27 @@ void BasePlaylistFeature::slotCreateImportPlaylist() {
     activatePlaylist(lastPlaylistId);
 }
 
+void BasePlaylistFeature::slotCreateImportPlaylistDirect(const QString& playlistFile) {
+    const QFileInfo fileInfo(playlistFile);
```

Comment:
QFileInfo is a heavy class, caching disk access. Please try to create it only once.